techmatters/terraso-mobile-client
# feat: Add sync button & handle deleted required entities on a few screens after pulling data from server

## Description
Spinoff work from #2230 
- Adds a sync button to most screens (behind feature flag) that queries for everything
- Update a few screens:
  - Use the new component wrapper `RestrictByRequirements`. It is intended to wrap screens and if any of its required data is missing, it will prevent the screen from rendering, and will do an action.
  - Make sure screens take id's and use redux to get data. If data is passed in as props, it may become stale if data is pulled from server while on that screen.
- Refactoring to break `LocationDashboardScreen` into `SiteTabsScreen` and `TemporaryLocationScreen`. Both will still use `LocationDashboardContent`. Some additional renaming in this area. Apologies, this is why there are many files with changes.

### Still to do for the "sync from server" work:
- How to test this?
- Do the right thing on screens when the data exists, but you shouldn't be allowed to see the current screen because permissions changed (this could happen if you were a manager editing a site note, then became a viewer).
- Audit all screens to specify requirements
- Identify when push response has an error, and then do a pull


### Related Issues
Implements #2232
